processor/src/trace/mod.rs

```diff
@@ -82,29 +82,38 @@ impl ExecutionTrace {
         // perfect zero knowledge.
         let program_hash: Digest = process.decoder.program_hash().into();
         let rng = RandomCoin::new(&program_hash.to_bytes());
+
+        // create a new program info instance with the underlying kernel
+        let kernel = process.kernel.clone();
```

Comment:
We should get `kernel` from `KernelRom` chiplet rather than from the process itself. This way, all kernel-related functionality is contained within `KernelRom`.

Reply:
Fixed!